src/MailProcessor.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+namespace Civi\Emailfiling;
+
+/**
+ * Class MailProcessor
+ *
+ * @package Civi\EmailFiling
+ */
+class MailProcessor {
+
+  /**
+   * Email data (comes from hook_civicrm_alterMailParams()).
+   *
+   * @var array
+   */
+  private $message;
+
+  /**
+   * EmlGenerator constructor.
+   *
+   * @param array $message
+   *   Email data (comes from hook_civicrm_alterMailParams()).
+   */
+  public function __construct(array $message) {
+    $message['timestamp'] = $message['timestamp'] ?? time();
+    $this->message = $message;
+  }
+
+  /**
+   * Attaches email as .eml file to activity.
+   *
+   * @param int $activityId
+   *   Activity id.
+   * @param string|null $eml
+   *   (Optional)Contents of eml file.
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+
+   * @return array|false
+   *   Attached file entity data, empty array or FALSE in case of error.
+   *
+   * @throws \CiviCRM_API3_Exception
+   *   An exception generated by civicrm_api3().
+   */
+  public function attachEmlToActivity($activityId, $eml = NULL, $message = NULL) {
```

Comment:
Do you think we can add one or two tests for this service? Test that an EML file is attached to the activity Id passed in and maybe other relevant tests.